File: mall-manager-web/src/main/webapp/js/service/goodsService.js
// service层,与数据库交互
app.service("goodsService",function($http){
    //BASE crud
    this.query=function(pageNum,pageSize,condition){
        return $http.post("../goods/query?pageNum="+pageNum+"&pageSize="+pageSize,condition);
    }

    this.save=function (goods) {
        return $http.post("../goods/save",goods);
    }

    this.remove=function (idList) {
        return $http.get("../goods/delete?idList="+idList);
    }

    this.queryById=function (id) {
        return $http.get("../goods/queryById?id="+id);
    }
});

File: mall-sellergoods-interface/src/main/java/com/marshal/sellergoods/service/GoodsService.java
package com.marshal.sellergoods.service;


import com.marshal.pojo.TbGoods;
import com.marshal.util.ResponseData;
import java.util.List;

public interface GoodsService {
	public List<TbGoods> findAll();

	public ResponseData query(TbGoods condition,int pageNum,int pageSize);

	public TbGoods queryById(Long id);

	public void save(TbGoods tbGoods);

	public void delete(Long[] ids);
}
